### ISSUE TITLE
Dimensional structure for datasets

### ISSUE THREAD

**Issue by KathiSchleidt** [FAIRiCUBE/data-requests#43]
I've been seeing all sorts of variants in the individual datasets being provided on rasdaman, some just have lat/long dimensions (although the dataset does have time aspects), other have lat/long/time (whereby only values for one time instant are being provided)

Apart from the confusion generated by such variance, it would be much easier to access the data if the structure is similar (e.g. then we know we must always slice for a single time instant if we want one value for a location, don't first have to look to see if this is even relevant for the dataset)

My proposal would be always using the 3 dimensions lat/long/time 

Thoughts?

**Comment by robknapen** [comment]
I agree that it should be standardised, perhaps even with some FAIRiCUBE (quality) guidelines so that users know what to expect from our data cubes.

But if we decide on geographic coordinates that needs to be clearly specified (so no X,Y coordinates), and for the temporal dimension we need to be clear on how we handle data with different temporal resolutions (hourly, daily, monthly, yearly, etc.).

Somehow we need to cross the gap between providing uncoordinated/raw EO data and Analysis Ready Data that is easy to use by researchers from the domains we want to support.

**Comment by pebau** [comment]
wait, geo data ARE multi-dimensional. Not sure what is confusing about that. Inventing extras does not make much sense - eg, if you add a time stamp then it means data are valid only at that times, whereas a DEM should always be applicable. Plus, you need to consider slicing if you want to get out 2D, which everybody will want.

Guess we have to accept that.

Likewise, the x/y stuff comes in different CRSs, that's just practice. 

It's life.

**Comment by pebau** [comment]
> for the temporal dimension we need to be clear on how we handle data with different temporal resolutions (hourly, daily, monthly, yearly, etc.).

fortunately, there is no magic: you just specify temporal resolution in the data request, we translate that into the ingestion syntax. We have services ranging in resolution from 3 years t 1 day to 6h to 5min.

**Comment by KathiSchleidt** [comment]
At present, Mohit has been struggling to provide data in yearly (or 3year) intervals. 
Temporal is where we need to find an agreement, can we create such a Year axis (pretty sure we can), as well as can we handle the ramifications (e.g. no WMS)?

As for the spatial dimensions, I thought we'd agreed for the moment to retain the original CRS, thus our coordinates will alternatively be lat/long and X/Y (or maybe N/E)

**Comment by pebau** [comment]
There is no such thing as a year axis, there is only time. The resolution parameter determines a regular time stepping; if there is no such  regularity we can define the axis as irregular, and timesteps can be at any distance.

Re spatial, yes, that is my understanding, too.

**Comment by KathiSchleidt** [comment]
@pebau To my understanding, @Mohinem issue was how to indicate years as he could only do days - ran into leap-year issues
Can one indicate that the steps are in years instead of days?

**Comment by pebau** [comment]
@KathiSchleidt help - what steps? There is only 2018 in my understanding.

**Comment by pebau** [comment]
@KathiSchleidt well, then please let us (= UC owners) document clearly. This example nicely shows how much is left unclear in the descriptions we receive, and we cannot yet read minds :)

**Comment by robknapen** [comment]
This might be part of the gap we try to cross between domains. Not every UC owner will know the exact spatial and temporal details and processing information that needs to be provided for ingestion. Even if you could read their minds, it would not be in there :-)

**Comment by pebau** [comment]
@robknapen wait, if the UC providers do not know their data, who then? Thought the data scientists are the experts on their data...in any case, we cannot find out what even the domain experts do not know. scratching my head, how can this be solved?

**Comment by robknapen** [comment]
@pebau Interesting question indeed. To me, UC owner, (UC) data provider, (UC) data scientists, and domain experts might sometimes be roles of the same person, but most likely it is different people and data scientists have to become experts (to the level required) of the data they are asked to analyse. With FAIRiCUBE we try to at least make their life a bit easier on the spatial-temporal idiosyncrasies of data via (gridded) data cubes.

So perhaps we need an intermediary (like a data steward) between data provider - data ‘ingestor’ (- data scientist)?

**Comment by pebau** [comment]
@robknapen good point indeed: We do it the hard way to make it easier for others. 

I have suggested to Kathi to hold a data briefing whenever a new request is filed to clarify all open questions. Possibly direct interaction can help a little.

**Comment by KathiSchleidt** [comment]
@pebau I believe we discussed the GitHub Issue functionality for this. Procedure in the case where the original request is missing some information would be to assign responsibles from the requesting use cases to the issue, clearly state the question in the issue. I have yet to see this happen. :(

On your statement that the data scientists MUST intimately know all datasets they're requesting, this is exactly the problem that FAIRiCUBE is aiming to resolve! No, the colleagues here at the Natural History Museum are not at all experts in the various data sources they'd need to complement their data, they only know what types of data would be useful in their further analysis (e.g. air temperature). The reason we set up the FAIRiCUBE project was that they knew that this data was available, but it remained sadly beyond their reach!

Our core task within FAIRiCUBE is to identify the various issues being encountered between knowing the data is there to actually being able to access and use the data. If the concepts provided in the Ingestion Requests are insufficient, I'd expect WP5 to update this form as required.

**Comment by KathiSchleidt** [comment]
While we're sorting the option of assigning data stewards, could we please progress on the actual content of this issue? When I look at the Coverages currently available on rasdaman, I see an eclectic mix of approaches, e.g.:

- near_surface_air_temperature: (daily values) uses ReferenceableGridByVectors, lists all available days
- corine_land_cover: (values every 3 years) uses RectifiedGrid specifying daily steps (but only provides data for 2018, so actually no temporal intervals)

Stability would be essential for the UC partners who are already trying to utilize this data!!!

**Comment by pebau** [comment]
closing as I believe the data requests should hosts specific issues whereas a general discussion can be continued over a beer.

**Comment by KathiSchleidt** [comment]
@pebau while the case-by-case approach is surely simpler for CU, a harmonized approach to data provision would be much more useful for UC partners

As recently discussed, I've analyzed the gridded datasets in the Inventory sheet (I created an additional sheet for this task, [Inventory_V2-Types](https://nilu365.sharepoint.com/:x:/r/sites/Horizon2021_CUBE/Shared%20Documents/WP2%20-%20use/DataLayerInventory/[Inventory_V2-Types.xlsx](https://nilu365.sharepoint.com/:x:/r/sites/Horizon2021_CUBE/Shared%20Documents/WP2%20-%20use/DataLayerInventory/Inventory_V2-Types.xlsx?d=w937b615a696d45f39be6155fd30619dc&csf=1&web=1&e=AYFuP6)?d=w937b615a696d45f39be6155fd30619dc&csf=1&web=1&e=AYFuP6)). I focused on the spatial and temporal resolution, together with the nature of the data (Classes, Percent, Measured Value, Modelled Value), in the 2nd Tab "Analysis", I described what types of data we have for each of the data nature types, think this sheet gives a good overview of the types of coverage encoding we'll require. Short summary here:
- CLS, CLC: tri-yearly data, 20m resolution until (incl.) 2015, 10m resolution from (incl.) 2018. EPSG:3035. Not interpolateable
- Climate Data: hourly (some daily, monthly, yearly, but should be discussed with UC). WGS84, 0.5° Grid. Interpolation depends on type of value and dimension, not possible to interpolate hourly values from daily averages
- EU Demography: 1km grid, EPSG:3035, data for 2018 and 2021. No interpolation

@robknapen I got lost on the Dutch national datasets, as not much access described. Do you see datasets from these sources which are different in structure from those sorted on the Analysis sheet?
